ci/jenkins/stop-jobs.sh

```diff
@@ -0,0 +1,151 @@
+}
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+    --pull-request)
+    PULL_REQUEST_ID="$2"
+    shift 2
+    ;;
+    --jenkins)
+    JENKINS_URL="$2"
+    shift 2
+    ;;
+    -h|--help)
+    print_usage
+    exit 0
+    ;;
+    *) 
+    echoerr "Unknown option $1"
+    exit 1
+    ;;
+esac
+done
+
+function abort_queue_jobs {
+    echo "====== checking for capv stale waiting jobs on PR ${PULL_REQUEST_ID} ======"
+    capv_job_name=$1
+    length_of_queue=$(curl -s -XGET "${JENKINS_URL}queue/api/json" | jq '[.[]][2]' | jq length)
+
+    for (( j=0; j<$length_of_queue; j++ )); do
+        job_name=$(curl -s -XGET "${JENKINS_URL}queue/api/json" | jq "[.[]][2][$j].task.name" | tr -d '"') 
+        if [[ "$job_name" == "$capv_job_name" ]]; then
+            # fetch the waiting job_id and waiting job pr_id
+            job_id=$(curl -s -XGET "${JENKINS_URL}queue/api/json" | jq "[.[]][2][$j].id")
+            waiting_job_pr_id=$(curl -s -XGET "${JENKINS_URL}queue/api/json" | jq "[.[]][2][$j].actions[1].parameters[9].value" | tr -d '"')
+
+            if [[ "$PULL_REQUEST_ID" == "$waiting_job_pr_id" ]]; then
+                echo "====== abort waiting job $capv_job_name on PR ${PULL_REQUEST_ID} ======"
+                curl -XPOST "${JENKINS_URL}/queue/cancelItem?id=$job_id" -u admin:$JENKINS_API_TOKEN
+                break
+            fi
+        fi
+    done
+}
+
+function abort_running_jobs {
+    echo "====== checking for capv stale running jobs on PR ${PULL_REQUEST_ID} ======"
+    capv_job_name=$1
+    i=$2
+    job_color=$(curl -s -XGET "${JENKINS_URL}label/antrea-test-node/api/json" | jq "[.[]][10][$i].color" | tr -d '"')
+
+    if [[ ${job_color: -5} == "anime" ]]; then
+        last_build_number=$(curl -s -XGET "${JENKINS_URL}job/$capv_job_name/api/json" | jq "[.[]][10][0].number" | tr -d '"')
+        pr_id=$(curl -s -XGET "${JENKINS_URL}job/"$capv_job_name"/"$last_build_number"/injectedEnvVars/api/json" | jq "[.[]][1].ghprbPullId"| tr -d '"')
+
+        if [[ "$PULL_REQUEST_ID" == "$pr_id" ]]; then
+            echo "====== abort running job $capv_job_name on PR ${PULL_REQUEST_ID} ======"
+
+            # need to stop first then clean
+            curl -XPOST "${JENKINS_URL}job/$capv_job_name/$last_build_number/stop" -u admin:$JENKINS_API_TOKEN
+
+            CLUSTER_NAME=$(curl -s -XGET "${JENKINS_URL}job/"$capv_job_name"/"$last_build_number"/injectedEnvVars/api/json" | jq "[.[]][1].BUILD_TAG"| tr -d '"')
+            ip_of_capv_testbed=$(curl -s -XGET "${JENKINS_URL}job/"$capv_job_name"/"$last_build_number"/injectedEnvVars/api/json" | jq "[.[]][1].SSH_CONNECTION" | awk '{print $3}')
+            
+            cleanup_running_job "$CLUSTER_NAME" "$ip_of_capv_testbed"
+        fi
+    fi   
+}
+
+function stop_kill() {
+    echo "====== checking for stale jobs on PR ${PULL_REQUEST_ID} ======"
+    length_of_capv_jobs=$(curl -s -XGET "${JENKINS_URL}label/antrea-test-node/api/json" | jq "[.[]][10]" | jq length)
```

Comment:
the query will return all jobs on a test node, In a single test bed there could multiple PR jobs might run with multiple kind cluster support !

Reply:
Hi ,actually here.. we are just fetching total number of jobs available on antrea-test-node so that we can check which capv (stale) job is running on particular PR.